cluster/service.go

```diff
@@ -74,22 +93,57 @@ func (c *Service) Open(ctx context.Context, db schema.Indexer) error {
 		return fmt.Errorf("open raft store: %w", err)
 	}
 
-	bs := bootstrap.NewBootstrapper(
-		c.rpcClient,
-		c.config.NodeID,
-		c.raftAddr,
-		c.config.NodeToAddressResolver,
-		c.Raft.Ready,
-	)
+	// If FQDN resolver is enabled make sure we're also using it for the bootstrapping process
+	nodeToAddressResolver := c.config.NodeToAddressResolver
+	if c.config.EnableFQDNResolver {
+		nodeToAddressResolver = resolver.NewFQDN(resolver.FQDNConfig{
+			// We don't need to specify more config as we are only using that resolver to resolve a node id to an IP
+			// overriding the default resolver using memberlist.
+			TLD: c.config.FQDNResolverTLD,
+		})
+	}
+
+	hasState, err := raft.HasExistingState(c.Raft.store.logCache, c.Raft.store.logStore, c.Raft.store.snapshotStore)
+	if err != nil {
+		return err
+	}
+	c.log.WithField("hasState", hasState).Info("raft init")
 
+	// If we have a state in raft, we only want to re-join the nodes in raft_join list to ensure that we update the
+	// configuration with our current ip.
+	// If we have no state, we want to do the bootstrap procedure where we will try to join a cluster or notify other
+	// peers that we are ready to form a new cluster.
 	bCtx, bCancel := context.WithTimeout(ctx, c.config.BootstrapTimeout)
 	defer bCancel()
-	if err := bs.Do(
-		bCtx,
-		c.config.NodeNameToPortMap,
-		c.logger,
-		c.config.Voter, c.closeBootstrapper); err != nil {
-		return fmt.Errorf("bootstrap: %w", err)
+	if hasState {
+		// Only join the remote nodes
+		bootstrapCtx, cancel := context.WithTimeout(ctx, c.config.BootstrapTimeout)
```

Comment:
**note:**  i think `bCtx` in line 116 does the same , maybe we just rename it for clearer name `bootstrapCtx` as the one you created 
```
	bCtx, bCancel := context.WithTimeout(ctx, c.config.BootstrapTimeout)
	defer bCancel()
```